xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-api/src/main/java/org/xwiki/ratings/internal/SolrRatingsManager.java

```diff
@@ -563,4 +565,31 @@ public AverageRating recomputeAverageRating(EntityReference entityReference) thr
             throw new RatingsException(AVERAGE_RATING_NOT_ENABLED_ERROR_MESSAGE);
         }
     }
+
+    private void checkIfVoteValid(int vote) throws RatingsException
+    {
+        // If the vote is outside the scope of the scale, we throw an exception immediately.
+        if (vote < 0 || vote > this.getScale()) {
+            throw new RatingsException(String.format("The vote [%s] is out of scale [%s] for [%s] rating manager.",
+                vote, this.getScale(), this.getIdentifier()));
+        }
+    }
+
+    private void checkIfDocumentExists(EntityReference reference) throws RatingsException
+    {
+        try {
+            if (!(reference instanceof DocumentReference documentReference)
```

Comment:
This feels a bit too reductive, it could target a document without being a DocumentReference. I would `#extractReference(EntityType.DOCUMENT)`, see if get something and if that EntityReference of type DOCUMENT exists.